apps/emqx_management/src/emqx_mgmt_api_subscriptions.erl

```diff
@@ -396,10 +401,21 @@ merge_queries(QString0, Q1, Q2) ->
             Q2Page = ceil(C1 / Limit),
             case Page =< Q2Page of
                 true ->
-                    #{data := Data, meta := #{hasnext := HN}} = Q1(QString0),
+                    #{data := Data1, meta := #{hasnext := HN1}} = Q1(QString0),
+                    NumRows1 = length(Data1),
+                    case NumRows1 >= Limit of
+                        true ->
+                            Data = Data1,
```

Comment:
this kind of variable bind is very confusing, it looks like assertion

suggestion: `{Data, HasNext} = case ... of`
or even better, write a func to handle the case part.

Reply:
:heavy_check_mark: